resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+package org.snomed.module.storage;
+
+import org.ihtsdo.otf.resourcemanager.ResourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snomed.otf.script.utils.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Perform basic operations of remotely stored RF2 packages in a controlled & consistent manner.
+ */
+public class ModuleStorageCoordinator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModuleStorageCoordinator.class);
+
+    private final ResourceManager resourceManager;
+    private final RF2Service rf2Service;
+    private final String writeDirectory;
+    private final List<String> readDirectories;
+    private final boolean allowArchive;
+
+    /**
+     * Constructor.
+     *
+     * @param resourceManager ResourceManager to use to communicate with either external or local storage system.
+     * @param rf2Service      RF2Service to use to process RF2 package.
+     * @param writeDirectory  Directory to write files to.
+     * @param readDirectories Collection of directories to read files from. If a file is not found in one of the read directories,
+     *                        then the next read directory will be checked.
+     * @param allowArchive    Control whether the archive method supported.
+     */
+    public ModuleStorageCoordinator(ResourceManager resourceManager, RF2Service rf2Service, String writeDirectory, List<String> readDirectories, boolean allowArchive) {
+        this.resourceManager = resourceManager;
+        this.rf2Service = rf2Service;
+        this.writeDirectory = writeDirectory;
+        this.readDirectories = readDirectories;
+        this.allowArchive = allowArchive;
+
+        if (readDirectories.size() == 1 && readDirectories.get(0).contains(",")) {
+            throw new IllegalArgumentException("Invalid read directories"); // e.g. List.of("a,b,c") instead of List.of("a", "b", "c")
+        }
+    }
+
+    /**
+     * Instantiate with Dev-environment configuration. This configuration allows for reading files from "dev" and "prod",
+     * but only writing to "dev".
+     *
+     * @param resourceManager ResourceManager to use to communicate with either external or local storage system.
+     * @return Instantiated class with Dev-environment configuration.
+     */
+    public static ModuleStorageCoordinator initDev(ResourceManager resourceManager) {
+        return new ModuleStorageCoordinator(resourceManager, new RF2Service(), "dev", List.of("dev", "prod"), true);
+    }
```

Comment:
@pgwilliams I've added 3 factory methods for easiness: initDev, initUat, initProd.